### DISCUSSION TITLE
Failed to deploy nextjs application that uses typescript to cloudflare

### DISCUSSION THREAD

**Discussion by byamasu-patrick** [vercel/next.js#40663] [Help]
Hello, I am trying to deploy a new nextjs application that uses react redux Tailwind CSS and Headless UI and Material UI to Cloudflare pages, but it is failing to install the node module (npm packages) is there anyone who would know how to solve this issue? I need to apply npm install --legacy-peer-deps so that the Cloudflare engine should be able to install all necessary packages but then it is not working

19:33:29.269 | Installing NPM modules using NPM version 8.15.0
-- | --
19:33:29.606 | npm WARN config tmp This setting is no longer used.  npm stores temporary files in a special
19:33:29.606 | npm WARN config location in the cache, and they are managed by
19:33:29.606 | npm WARN config     [`cacache`](http://npm.im/cacache).
19:33:29.942 | npm WARN config tmp This setting is no longer used.  npm stores temporary files in a special
19:33:29.943 | npm WARN config location in the cache, and they are managed by
19:33:29.943 | npm WARN config     [`cacache`](http://npm.im/cacache).
19:33:31.134 | npm ERR! code ERESOLVE
19:33:31.138 | npm ERR! ERESOLVE could not resolve
19:33:31.139 | npm ERR!
19:33:31.139 | npm ERR! While resolving: @material-ui/core@4.12.4
19:33:31.139 | npm ERR! Found: @types/react@18.0.17
19:33:31.139 | npm ERR! node_modules/@types/react
19:33:31.139 | npm ERR!   dev @types/react@"18.0.17" from the root project
19:33:31.139 | npm ERR!   peerOptional @types/react@"*" from @material-ui/types@5.1.0
19:33:31.139 | npm ERR!   node_modules/@material-ui/types
19:33:31.139 | npm ERR!     @material-ui/types@"5.1.0" from @material-ui/core@4.12.4
19:33:31.139 | npm ERR!     node_modules/@material-ui/core
19:33:31.140 | npm ERR!       peer @material-ui/core@"^4.0.0" from @material-ui/icons@4.11.3
19:33:31.140 | npm ERR!       node_modules/@material-ui/icons
19:33:31.140 | npm ERR!         @material-ui/icons@"^4.11.3" from the root project
19:33:31.140 | npm ERR!       1 more (@types/mui-datatables)
19:33:31.140 | npm ERR!     @material-ui/types@"5.1.0" from @material-ui/styles@4.11.5
19:33:31.140 | npm ERR!     node_modules/@material-ui/styles
19:33:31.140 | npm ERR!       @material-ui/styles@"^4.11.5" from the root project
19:33:31.140 | npm ERR!       1 more (@material-ui/core)
19:33:31.140 | npm ERR!     1 more (@types/mui-datatables)
19:33:31.141 | npm ERR!   12 more (@mui/base, @mui/icons-material, @mui/material, ...)
19:33:31.141 | npm ERR!
19:33:31.141 | npm ERR! Could not resolve dependency:
19:33:31.141 | npm ERR! peerOptional @types/react@"^16.8.6 \|\| ^17.0.0" from @material-ui/core@4.12.4
19:33:31.141 | npm ERR! node_modules/@material-ui/core
19:33:31.141 | npm ERR!   peer @material-ui/core@"^4.0.0" from @material-ui/icons@4.11.3
19:33:31.141 | npm ERR!   node_modules/@material-ui/icons
19:33:31.141 | npm ERR!     @material-ui/icons@"^4.11.3" from the root project
19:33:31.141 | npm ERR!   @material-ui/core@"^4.11.3" from @types/mui-datatables@3.7.9
19:33:31.141 | npm ERR!   node_modules/@types/mui-datatables
19:33:31.141 | npm ERR!     @types/mui-datatables@"^3.7.9" from the root project
19:33:31.142 | npm ERR!
19:33:31.142 | npm ERR! Conflicting peer dependency: @types/react@17.0.50
19:33:31.142 | npm ERR! node_modules/@types/react
19:33:31.142 | npm ERR!   peerOptional @types/react@"^16.8.6 \|\| ^17.0.0" from @material-ui/core@4.12.4
19:33:31.142 | npm ERR!   node_modules/@material-ui/core
19:33:31.142 | npm ERR!     peer @material-ui/core@"^4.0.0" from @material-ui/icons@4.11.3
19:33:31.142 | npm ERR!     node_modules/@material-ui/icons
19:33:31.142 | npm ERR!       @material-ui/icons@"^4.11.3" from the root project
19:33:31.142 | npm ERR!     @material-ui/core@"^4.11.3" from @types/mui-datatables@3.7.9
19:33:31.142 | npm ERR!     node_modules/@types/mui-datatables
19:33:31.143 | npm ERR!       @types/mui-datatables@"^3.7.9" from the root project
19:33:31.143 | npm ERR!
19:33:31.143 | npm ERR! Fix the upstream dependency conflict, or retry
19:33:31.143 | npm ERR! this command with --force, or --legacy-peer-deps
19:33:31.143 | npm ERR! to accept an incorrect (and potentially broken) dependency resolution.
19:33:31.143 | npm ERR!
19:33:31.143 | npm ERR! See /opt/buildhome/.npm/eresolve-report.txt for a full report.
19:33:31.143 |  
19:33:31.143 | npm ERR! A complete log of this run can be found in:
19:33:31.143 | npm ERR!     /opt/buildhome/.npm/_logs/2022-09-18T17_33_29_917Z-debug-0.log
19:33:31.151 | Error during NPM install
19:33:31.154 | Failed: build command exited with code: 1
19:33:31.959 | Failed: an internal error occurred

**Comment by mohdanss** [comment]
It's probably late, but deleting the `package-lock.json` file resolved my issue.
